src/public/modules/forms/base/directives/submit-form.directive.js

```diff
@@ -101,7 +102,25 @@ function submitFormDirective(
         if (isPersistentLogin) GTag.persistentLoginUse(scope.form)
 
         const query = $location.url().split('?')
-        const encodedQuery = query.length > 1 ? btoa(query[1]) : undefined
+        const queryString = query.length > 1 ? query[1] : undefined
+        const queryId = queryString ? new ObjectId() : undefined
+        const encodedQuery = queryId ? btoa(`queryId=${queryId}`) : undefined
+        const queryObject = {
+          _id: queryId,
+          queryString,
+        }
+
+        if (queryString) {
+          // Defensive - try catch block in case the storage is full
+          // See https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem
+          try {
+            localStorage.setItem('storedQuery', JSON.stringify(queryObject))
```

Comment:
what are your considerations behind using localStorage instead of sessionStorage? I imagine the latter would work too as the browser/tab remains the same throughout the login, and has the advantage of not persisting data which could subsequently be stolen

Reply:
thats a fair point - i used localStorage as I considered the possibility that the user might open the singpass login in a new tab in which case it would be a different session, but I realize that our app doesn't allow that anyway since login is by way of redirect, so sessionStorage it will be then